apps/zero-runtime-vite-app/src/App.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react';
+import { styled } from '@mui/zero-runtime';
+import Slider from './Slider/ZeroSlider';
+
+const Button = styled('button', {
+  name: 'MuiButton',
+  slot: 'Root',
+})(
+  () => ({
+    fontFamily: 'sans-serif',
+    '--var-tag': 'h2',
+    color: 'primary.main',
+    backgroundColor: ['primary.main', 'text.primary', 'background.paper'],
+  }),
+  {
+    fontFamily: 'sans-serif',
+    '--var-tag': 'h1',
+    // p: (props: any) => (props.isRed ? 10 : 20),
+    color: (props: any) => (props.isRed ? 'primary.main' : 'secondary.main'),
```

Comment:
After discussion, the theem is not supported in the props callback, we should document this.